spring-integration-zookeeper/src/test/java/org/springframework/integration/zookeeper/ZookeeperTestSupport.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.zookeeper;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.BoundedExponentialBackoffRetry;
+import org.apache.curator.test.TestingServer;
+import org.apache.curator.utils.CloseableUtils;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public class ZookeeperTestSupport {
+
+	protected final Log log = LogFactory.getLog(this.getClass());
+
+	private TestingServer testingServer;
+
+	protected CuratorFramework client;
+
+	public ZookeeperTestSupport() {
+		super();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		testingServer = new TestingServer(true);
+		client = createNewClient();
+	}
+
+	@After
+	public void tearDown() {
+		closeClient(this.client);
+		try {
+			testingServer.stop();
+		}
+		catch (IOException e) {
+			log.warn("Exception thrown while shutting down ZooKeeper: ", e);
+		}
+		testingServer.getTempDirectory().delete();
+	}
```

Comment:
From other side in the real world we really get deal with shared instances and make objects clean and fresh is really a task for the end-application. 

I'd be happy do not have any issues with a shared server, but right now we doesn't reflect that with current solution.

Well, to be kind we really can investigate that as a separate JIRA, but you haven't convinced me that your solution works well in distributed shared environment, because you provide only instance-per-task state. And I really sure that tidying server in the `@Before` would be better by performance than new server for each test.

Having those objections, I'd say that PR isn't acceptable yet, so no reason to move rest task to the separate JIRA :smile: .